lib/SILOptimizer/SILCombiner/SILCombine.cpp

```diff
@@ -152,6 +153,44 @@ bool SILCombiner::doOneIteration(SILFunction &F, unsigned Iteration) {
 
   SILCombineCanonicalize scCanonicalize(Worklist);
 
+  // Match test program:
```

Comment:
Here's the example from the description. I'll remove this before committing, but I wanted to have a _working_ example without implementing an actual optimization (in the same patch).